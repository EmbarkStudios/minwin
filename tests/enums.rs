mod utils;
use utils::*;

// Ensures we can emit some but not all of a simple enum's values
test!(
    simple,
    [
        BindConfig::Minwin(Default::default()),
        BindConfig::Minwin(MinwinBindConfig {
            enum_style: EnumStyle::Minwin,
            ..Default::default()
        }),
        BindConfig::Minwin(MinwinBindConfig {
            enum_style: EnumStyle::Minwin,
            use_rust_casing: true,
            ..Default::default()
        }),
        BindConfig::Bindgen,
    ],
    [
        "Windows.Win32.Foundation.GENERIC_READ",
        "Windows.Win32.Foundation.GENERIC_WRITE",
        "Windows.Win32.Foundation.GENERIC_ALL",
        "Windows.Win32.Storage.FileSystem.FILE_TYPE_DISK",
        "Windows.Win32.Storage.FileSystem.GetFileType",
        "Windows.Win32.Storage.FileSystem.ClfsMgmtLogWriteNotification",
        "Windows.Win32.Storage.FileSystem.ReadLogNotification",
    ]
);

// Ensures we can emit enums that use core types instead of a typedef
test!(
    core,
    [
        BindConfig::Minwin(Default::default()),
        BindConfig::Minwin(MinwinBindConfig {
            enum_style: EnumStyle::Minwin,
            ..Default::default()
        }),
        BindConfig::Minwin(MinwinBindConfig {
            enum_style: EnumStyle::Minwin,
            use_rust_casing: true,
            ..Default::default()
        }),
        BindConfig::Bindgen,
    ],
    ["Windows.Win32.Foundation.S_OK"]
);

// Ensures we can emit the enum type alias, even if no actual enum variants
// are requested
test!(
    always_emits_alias,
    [
        BindConfig::Minwin(Default::default()),
        BindConfig::Minwin(MinwinBindConfig {
            enum_style: EnumStyle::Minwin,
            ..Default::default()
        }),
        BindConfig::Minwin(MinwinBindConfig {
            enum_style: EnumStyle::Minwin,
            use_rust_casing: true,
            ..Default::default()
        }),
        BindConfig::Bindgen,
    ],
    ["Windows.Win32.UI.Input.Pointer.GetPointerTouchInfo"]
);

// Ensures that enums that clash with handle types are supported
test!(
    handle_and_enum,
    [
        BindConfig::Minwin(Default::default()),
        BindConfig::Minwin(MinwinBindConfig {
            enum_style: EnumStyle::Minwin,
            ..Default::default()
        }),
        BindConfig::Minwin(MinwinBindConfig {
            enum_style: EnumStyle::Minwin,
            use_rust_casing: true,
            ..Default::default()
        }),
        BindConfig::Bindgen,
    ],
    [
        "Windows.Win32.UI.WindowsAndMessaging.SetWindowPos",
        "Windows.Win32.UI.WindowsAndMessaging.GetMessageW",
        "Windows.Win32.UI.WindowsAndMessaging.HWND_NOTOPMOST",
    ]
);
