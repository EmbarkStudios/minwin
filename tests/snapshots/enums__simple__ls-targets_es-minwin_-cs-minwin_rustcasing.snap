---
source: tests/enums.rs
expression: bo.bindings
---
#![allow(
    non_snake_case,
    non_upper_case_globals,
    non_camel_case_types,
    clippy::upper_case_acronyms
)]
::windows_targets::link!(
    "clfsw32.dll" "system" "ReadLogNotification" fn read_log_notification(h_log : Handle,
    p_notification : * mut ClfsMgmtNotification, lp_overlapped : * mut Overlapped) ->
    Bool
);
::windows_targets::link!(
    "kernel32.dll" "system" "GetFileType" fn get_file_type(h_file : Handle) ->
    FileType::Enum
);
pub type Bool = i32;
#[repr(C)]
pub struct ClfsMgmtNotification {
    pub notification: ClfsMgmtNotificationType::Enum,
    pub lsn: ClsLsn,
    pub log_is_pinned: u16,
}
pub mod ClfsMgmtNotificationType {
    pub type Enum = i32;
    pub const ClfsMgmtLogWriteNotification: Enum = 3;
}
#[repr(C)]
pub struct ClsLsn {
    pub internal: u64,
}
pub mod FileType {
    pub type Enum = u32;
    pub const FileTypeDisk: Enum = 1;
}
pub mod GenericAccessRights {
    pub type Enum = u32;
    pub const GenericAll: Enum = 268435456;
    pub const GenericWrite: Enum = 1073741824;
    pub const GenericRead: Enum = 2147483648;
}
pub type Handle = isize;
#[repr(C)]
pub struct Overlapped {
    pub internal: usize,
    pub internal_high: usize,
    pub anonymous: Overlapped_0,
    pub h_event: Handle,
}
#[repr(C)]
pub union Overlapped_0 {
    pub anonymous: ::std::mem::ManuallyDrop<Overlapped_0_0>,
    pub pointer: *mut ::core::ffi::c_void,
}
#[repr(C)]
pub struct Overlapped_0_0 {
    pub offset: u32,
    pub offset_high: u32,
}

